x/authz/codec/codec.go

```diff
@@ -4,26 +4,26 @@ import (
 	types "github.com/cosmos/cosmos-sdk/codec/types"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	"github.com/cosmos/cosmos-sdk/types/msgservice"
-	"github.com/cosmos/cosmos-sdk/x/authz/exported"
+	"github.com/cosmos/cosmos-sdk/x/authz"
 	bank "github.com/cosmos/cosmos-sdk/x/bank/types"
 	staking "github.com/cosmos/cosmos-sdk/x/staking/types"
 )
 
 // RegisterInterfaces registers the interfaces types with the interface registry
 func RegisterInterfaces(registry types.InterfaceRegistry) {
 	registry.RegisterImplementations((*sdk.MsgRequest)(nil),
-		&MsgGrantAuthorizationRequest{},
-		&MsgRevokeAuthorizationRequest{},
-		&MsgExecAuthorizedRequest{},
+		&authz.MsgGrant{},
+		&authz.MsgRevoke{},
+		&authz.MsgExec{},
 	)
 
 	registry.RegisterInterface(
 		"cosmos.authz.v1beta1.Authorization",
-		(*exported.Authorization)(nil),
+		(*authz.Authorization)(nil),
 		&bank.SendAuthorization{},
-		&GenericAuthorization{},
+		&authz.GenericAuthorization{},
 		&staking.StakeAuthorization{},
```

Comment:
Woahh... we should not be registering `bank` and `staking` authorizations here. These should be registered by the bank and staking modules. Then we can move this out of it's own package and back into `x/authz`

Reply:
totally, I will move it.